ui/src/alerting/components/notifications/EditRuleOverlay.tsx

```diff
@@ -0,0 +1,114 @@
+// Libraries
+import React, {FC, useReducer, ChangeEvent} from 'react'
+import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
+
+// Components
+import RuleSchedule from 'src/alerting/components/notifications/RuleSchedule'
+import RuleConditions from 'src/alerting/components/notifications/RuleConditions'
+import RuleMessage from 'src/alerting/components/notifications/RuleMessage'
+import {
+  Panel,
+  ComponentSize,
+  Overlay,
+  Form,
+  Input,
+  Grid,
+  Columns,
+} from '@influxdata/clockface'
+
+// Reducers
+import {reducer, ActionPayload} from './RuleOverlay.reducer'
+
+// Constants
+import {endpoints} from 'src/alerting/constants' // Hooks
+import {RuleModeContext, EditRuleDispatch, RuleMode} from 'src/shared/hooks'
+
+// Types
+import {NotificationRuleDraft, AppState} from 'src/types'
+
+interface StateProps {
+  stateRule: NotificationRuleDraft
+}
+
+type Props = WithRouterProps & StateProps
+
+const EditRuleOverlay: FC<Props> = ({params, router, stateRule}) => {
+  if (!stateRule) {
+    return null
+  }
+
+  const handleDismiss = () => {
+    router.push(`/orgs/${params.orgID}/alerting`)
+  }
+
+  const mode = RuleMode.Edit
+  const [rule, dispatch] = useReducer(reducer(mode), stateRule)
```

Comment:
I wonder if it matters that this reducer is recreated every time this component renders.

Only time this bit me in the past is when I tried to create an impure reducer function. When the identity of a reducer changes from render to render, react will sometimes run the reducer twice for a single action, which tripped me up.